packages/chat-component-bindings/src/messageThreadSelector.ts

```diff
@@ -148,26 +149,42 @@ const extractAttachmentUrl = (attachment: ChatAttachment): string => {
   return attachment.url || '';
 };
 const processChatMessageContent = (message: ChatMessageWithStatus): string | undefined => {
+  let content = message.content?.message;
   /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
-  if (sanitizedMessageContentType(message.type).includes('html') && message.content?.attachments) {
+  if (content && sanitizedMessageContentType(message.type).includes('html') && message.content?.attachments) {
     const attachments: ChatAttachment[] = message.content?.attachments;
+    // Fill in the src here
+    /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+    const document = new DOMParser().parseFromString(content, 'text/html');
+    /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+    document.querySelectorAll('img').forEach((img) => {
+      const attachmentPreviewUrl = attachments.find((attachment) => attachment.id === img.id)?.previewUrl;
+      if (attachmentPreviewUrl) {
+        const src = message.resourceCache?.[attachmentPreviewUrl] ?? '';
+        img.src = src;
+      }
+    });
+    /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+    content = document.documentElement.innerHTML;
     const teamsImageHtmlContent = attachments
       .filter(
         (attachment) => attachment.attachmentType === 'image' && !message.content?.message?.includes(attachment.id)
       )
-      .map((attachment) => generateImageAttachmentImgHtml(attachment))
+      .map((attachment) => generateImageAttachmentImgHtml(message, attachment))
       .join('');
     if (teamsImageHtmlContent) {
-      return (message.content?.message ?? '') + teamsImageHtmlContent;
+      return (content ?? '') + teamsImageHtmlContent;
     }
   }
-  return message.content?.message;
+  return content;
 };
 
 /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
-const generateImageAttachmentImgHtml = (attachment: ChatAttachment): string => {
+const generateImageAttachmentImgHtml = (message: ChatMessageWithStatus, attachment: ChatAttachment): string => {
   const contentType = extractAttachmentContentTypeFromName(attachment.name);
-  return `\r\n<p><img alt="image" src="" itemscope="${contentType}" id="${attachment.id}"></p>`;
+  const previewUrl = attachment.previewUrl ?? '';
```

Comment:
shouldn't we abort creating image tag if previewUrl is empty? I think we should do nothing if it's missing